clients/sled-agent-client/src/lib.rs

```diff
@@ -52,25 +53,62 @@ impl Eq for types::OmicronZoneConfig {}
 impl Eq for types::OmicronZoneType {}
 impl Eq for types::OmicronZoneDataset {}
 
+/// Like [`types::OmicronZoneType`], but without any associated data.
+///
+/// The [`fmt::Display`] impl for this type is a human-readable label, meant
+/// for testing and reporting.
+#[derive(Debug, Clone, Copy, PartialEq, Eq, Hash, PartialOrd, Ord)]
+pub enum ZoneTag {
```

Comment:
I have also wanted something like this. FWIW we also have `nexus_db_model::ServiceKind`, which is _almost_ the same (it has cases for a couple services within the switch zone, but not all, and doesn't distinguish boundary from internal NTP). I wonder if those can be unified. I don't think that fits in the scope of this PR, but curious if you have thoughts.

Reply:
Yeah I did find that enum. This felt like it was doing almost the same thing, but not quite, and I'm not entirely sure how to unify the two.

Maybe we should move `ZoneTag` next to `ServiceType` though, and document the differences, to make future archeologists wonder less.

Reply:
What do you think about calling this `Kind`?  That's what we call it elsewhere and "tag" could mean a lot of different hings.  ("kind" can too but at least I'd expect a thing to have only one "kind", whereas it may have many tags.)

According to dbinit.sql, the database `service_kind` enum ought to be used only for the services table, which I think does not contain switch zone services (this is from dogfood):

```
> select distinct kind from service;
       kind
-------------------
  clickhouse
  cockroach
  crucible
  crucible_pantry
  external_dns
  internal_dns
  nexus
  ntp
  oximeter
(9 rows)


```

and the certificates table (which I think currently only supports Nexus as the service kind).  Maybe we can drop `ServiceKind` when we drop the `services` table and be left with just this one?

Reply:
Good point about kind vs tag -- I also think of entities as potentially having many tags.

I was hesitating to call it `ZoneKind` because of `ServiceKind`. And I saw a survey on some social media (maybe Fediverse) a while ago where Rust devs were being polled for how they name parallel enums without associated data -- `Kind` and `Tag` were the top two results. But if we just drop `ServiceKind` that becomes moot and `ZoneKind` is better.

Reply:
Changed this to `ZoneKind.